### ISSUE TITLE
Accessibility: Compass not properly working for all screenreader or keyboard-only users

### ISSUE THREAD

**Issue by behofer** [mongodb-js/compass#2776]
## General description
Visually impaired users like myself rely on a screenreader to have information read by a speech synthesizer or displayed on a braille display. Popular screenreaders are VoiceOver on Mac OS as well as JAWS and NVDA on Windows or Orca on Linux Desktop.
Currently a screenreader isn't able to work properly with MongoDB Compass.

## Steps to reproduce
* Download or use a free screenreader like [NVDA](https://www.nvaccess.org/) (Windows), [Microsoft Narrator(https://support.microsoft.com/en-us/windows/complete-guide-to-narrator-e4397a0d-ef4f-b386-d8ae-c172f109bdb1) (Windows) or [VoideOver(https://support.apple.com/guide/voiceover/vo2682/mac)
* Launch Compass and connect to a database
* Navigate Compass, e.g. explore the tree displaying your databases and collections, try to interact with elements on screen
* You will notice that screenreaders will announce buttons without labels, missing or not clearly labeled navigation elements as well as overall keyboard navigation problems

## Actual behaviour
A screenreader doesn't read elements properly. Symbolic buttons are not labeled which makes it impossible for the user to find out their purpose. Navigation elements like trees are not navigatable by keyboard. ...

## Expected behaviour
Compass is accessible for screenreader users as well as keyboard-only users.

## Suggestions
It's not necessarily about fully implementing [Web Content Accessibility Guidelines(https://www.w3.org/WAI/standards-guidelines/wcag/). A screenreader user will be helped by labeling elements correctly using HTML/JavaScript, using [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) and having navigational elements accessible for a first step.

## A few resources
* [React Accessibility Tools – How to Build More Accessible React Apps](https://www.freecodecamp.org/news/react-accessibility-tools-build-accessible-react-apps/)
* [Making React Apps More Accessible | DigitalOcean](https://www.google.com/search?q=example+accessible+react+app&rlz=1C1GCEA_en&oq=example+accessible+react+app&aqs=chrome..69i57j33i22i29i30l7.5593j0j4&sourceid=chrome&ie=UTF-8)
* [Accessibility in React(https://reactjs.org/docs/accessibility.html)
* [Accessibility in React (Mozilla article)](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/React_accessibility)
* [Making trees navigatable with keyboard (ARIA tree)](https://www.w3.org/WAI/GL/wiki/Using_ARIA_trees)
